src/WebJobs.Script.WebHost/Standby/StandbyManager.cs

```diff
@@ -51,6 +53,13 @@ public Task SpecializeHostAsync()
 
         public async Task SpecializeHostCoreAsync()
         {
+            _logger.LogInformation(Resources.HostSpecializationTrace);
+
+            // After specialization, we need to ensure that custom timezone
+            // settings configured by the user (WEBSITE_TIME_ZONE) are honored.
+            // DateTime caches timezone information, so we need to clear the cache.
+            TimeZoneInfo.ClearCachedData();
```

Comment:
When comparing to v1, I noticed we missed this. I assumed it was unintentional so I re-added it.

Reply:
Absolutely unintentional! thanks for catching that.

Reply:
Might consider adding a simple check to the standby mode test that causes TimeZoneInfo to cache, then verifies (likely via private reflection) that the cache was cleared. Should have had that test before, since this is a subtle issue that was reported by customers.

Reply:
I've added a check that the internal cache field changes during specialization. With this commented out, the test fails.